sdv/data_processing/data_processor.py

```diff
@@ -103,6 +104,112 @@ def __init__(self, metadata, enforce_rounding=True, enforce_min_max_values=True,
+            list:
+                A list of columns that are part of the address transformers.
+        """
+        try:
+            self._check_import_address_transformers()
+            return [
+                item for col_tuple, transformer in self.columns_to_multi_col_transformer.items()
+                if isinstance(
+                    transformer, (
+                        rdt.transformers.RandomLocationGenerator,
+                        rdt.transformers.RegionalAnonymizer
+                    )
+                ) for item in col_tuple
+            ]
+        except ImportError:
+            return []
+
+    def _get_address_transformer_parameters(self, column_names):
+        """Get the parameters for the address transformer.
+
+        Args:
+            column_names (tuple[str]):
+                The column names to get the parameters for.
+
+        Returns:
+            dict:
+                A dictionary mapping column names to sdtypes.
+        """
+        columns_to_sdtypes = {}
+
+        for column in column_names:
+            sdtype = self.metadata.columns[column]['sdtype']
+            columns_to_sdtypes[column] = sdtype
+
+        return columns_to_sdtypes
+
+    def _get_address_transformer(self, anonymization_level):
+        """Get the address transformer.
+
+        Args:
+            anonymization_level (str):
+                The anonymization level for the address transformer.
+        """
+        self._check_import_address_transformers()
+        transformer = None
+        if anonymization_level == 'full':
+            transformer = rdt.transformers.RandomLocationGenerator()
+        elif anonymization_level == 'street_address':
+            transformer = rdt.transformers.RegionalAnonymizer()
+
+        return transformer
+
+    def _update_address_transformer(self, transformer, columns_to_sdtypes):
+        """Update the transformer parameters.
+
+        Args:
+            transformer (rdt.transformers):
+                The transformer to update.
+            columns_to_sdtypes (dict):
+                A dictionary mapping column names to sdtypes.
+        """
+        list_sdtypes = list(columns_to_sdtypes.values())
+        transformer.locales = self._locales
+        transformer.columns_to_sdtypes = columns_to_sdtypes
+        transformer._list_sdtypes = list_sdtypes
+
+    def _set_address_transformer(self, column_names, anonymization_level):
+        """Set the address transformer.
+
+        Args:
+            column_names (tuple[str]):
+                The column names to set the transformer for.
+            anonymization_level (str):
+                The anonymization level for the address transformer.
+        """
+        columns_to_sdtypes = self._get_address_transformer_parameters(column_names)
+        transformer = self._get_address_transformer(anonymization_level)
+        self._update_address_transformer(transformer, columns_to_sdtypes)
+        transformer._validate_sdtypes()
```

Comment:
Both of this are not needed

Reply:
I know that a lot of logic here will be done by the `HyperTransformer`, but I need them because in the Issue we want to do some `validations` directly when calling `set_address_columns()`, so before creating any config or setting the `HyperTransformer`. If it's fine to let all the validation in a second time to the HyperTransformer, I agree we can remove those methods.

Reply:
Yes I see that, but this is just bad design then and requires changes to the transformer itself in order for it to handle everything. Basically `validate_sdtypes` could be a `classmethod` that takes the `columns_to_sdtypes` that you want to validate or the transformer itself takes those as input on instantiation and you are able to validate them with this same method.

The problem I'm having with this is that we are setting things on the transformer which ends up working like a mock, and those attributes are due to change further in the execution flow (during `fit` for example), that is why I'm a against the `transformer.attribute = new_value`, you can see that this is also more easy to get a bug than doing `Transformer(new_value)` or `transformer.update_attribute(new_value)` as per `transformer._locales` instead of `transformer.locales`.